array_api_tests/test_creation_functions.py

```diff
@@ -77,9 +78,28 @@ def test_empty(shape, dtype):
         shape = (shape,)
     assert a.shape == shape, "empty() produced an array with an incorrect shape"
 
-# TODO: implement empty_like (requires hypothesis arrays support)
-def test_empty_like():
-    pass
+
+@given(
+    a=xps.arrays(
+        dtype=shared_dtypes,
+        shape=xps.array_shapes(),
+    ),
+    dtype=one_of(none(), shared_dtypes.flatmap(promotable_dtypes)),
```

Comment:
It's possible I'm wrong here (flatmap always confuses me), but can't this (and elsewhere) just be `promotable_dtypes(shared_dtypes)`?

Reply:
Ah I wrote to `promotable_dtypes` to only except a single dtype, not a dtype-generating strategy. Actually I'll make it accept both in it's method body.